File: src/week3/P005.py
# 1931 회의실 배정
import sys
input = sys.stdin.readline

n = int(input())
meeting_times = []

for _ in range(n):
    start, end = map(int, input().split())
    meeting_times.append((start, end))

# 회의 시간을 종료 시간 기준 및 시작 시간 기준으로 정렬
meeting_times = sorted(meeting_times, key = lambda x: (x[1], x[0]))

count = 1
# 이전 회의 종료 시간
end_time = meeting_times[0][1]

for i in range(1, n):
    # 다음 회의 시작 시간이 이전 회의 종료 시간보다 크거나 같다면
    if (meeting_times[i][0] >= end_time):
        end_time = meeting_times[i][1]
        count += 1

print(count)

File: src/week3/P005.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0]);
const meetingTimes = [];

for (let i = 1; i <= n; i++) {
    meetingTimes.push(input[i].split(' ').map(Number));
}

// 회의 시간을 정렬 (핵심 로직)
// 1. 종료 시간을 기준으로 오름차순 정렬
// 2. 종료 시간이 같다면, 시작 시간을 기준으로 오름차순 정렬
meetingTimes.sort((a, b) => {
    if (a[1] !== b[1]) {
        return a[1] - b[1];
    } else {
        return a[0] - b[0];
    }
});

let count = 0;
let prevEndTime = 0;

for (const meeting of meetingTimes) {
    const startTime = meeting[0];
    const endTime = meeting[1];

    // 이전 회의 종료 시간보다 현재 회의 시작 시간이 크거나 같다면
    if (startTime >= prevEndTime) {
        // 해당 회의를 선택하고, 종료 시간을 갱신
        prevEndTime = endTime;
        count++;
    }
}

console.log(count);